app/models/rule.rb

```diff
@@ -33,10 +33,20 @@ class Rule < ApplicationRecord
   has_many :hosts, through: :rule_results, source: :host
   has_many :rule_references_rules, dependent: :delete_all
   has_many :rule_references, through: :rule_references_rules
+  has_many :required_for_rule_group_rules, class_name: 'RuleGroupRule',
+                                           as: :requiring,
+                                           dependent: :delete_all
+  has_many :conflicted_with_rule_group_rules, class_name: 'RuleGroupRule',
```

Comment:
Sounds better in my head, but please tell me if I'm contradicting a previous suggestion:
```suggestion
  has_many :conflicting_rule_group_rules, class_name: 'RuleGroupRule',
```

Reply:
I had this at first but I changed it to `conflicted_with_rule_group_rules` because if you were to call this association on a rule group or rule, it'd return all of the rule groups or rules that has this rule or rule group as a conflict. Does this make sense? I could change it to `conflicting_with_rule_group_rules`, maybe that would sound better? Or should I just use `conflicting_rule_group_rules`?

Reply:
Okay, sounds good! Yeah, these are polymorphic